mr/src/test/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandlerBaseTest.java

```diff
@@ -157,6 +177,282 @@ public void testJoinTables() throws IOException {
+          "Table does not exist", () -> {
+            Catalogs.loadTable(shell.getHiveConf(), properties);
+          }
+      );
+    } else {
+      // Check the HMS table parameters
+      Path hmsTableLocation;
+      try {
+        client = new HiveMetaStoreClient(metastore.hiveConf());
+        hmsTableLocation = new Path(client.getTable("default", "customers").getSd().getLocation());
+      } finally {
+        if (client != null) {
+          client.close();
+        }
+      }
+
+      // Drop the table
+      shell.executeStatement("DROP TABLE customers");
+
+      // Check if we drop an exception when trying to drop the table
+      AssertHelpers.assertThrows("should throw exception", NoSuchTableException.class,
+          "Table does not exist", () -> {
+            Catalogs.loadTable(shell.getHiveConf(), properties);
+          }
+      );
+
+      // Check if the files are kept
+      FileSystem fs = Util.getFs(hmsTableLocation, shell.getHiveConf());
+      // TODO: files should be deleted
+      // Assert.assertEquals(0, fs.listStatus(hmsTableLocation).length);
+    }
+  }
+
+  @Test
+  public void testCreateTableWithoutSpec() throws TException {
+    // We need the location for HadoopTable based tests only
+    String location = locationForCreateTable(temp.getRoot().getPath(), "customers");
+    shell.executeStatement("CREATE EXTERNAL TABLE customers " +
+        "STORED BY 'org.apache.iceberg.mr.hive.HiveIcebergStorageHandler' " +
+        (location != null ? "LOCATION '" + location + "' " : "") +
+        "TBLPROPERTIES ('" + InputFormatConfig.TABLE_SCHEMA + "'='" + SchemaParser.toJson(CUSTOMER_SCHEMA) + "')");
+
+    Properties properties = new Properties();
+    properties.put(Catalogs.NAME, TableIdentifier.of("default", "customers").toString());
+    if (location != null) {
+      properties.put(Catalogs.LOCATION, location);
+    }
+
+    // Check the Iceberg table partition data
+    org.apache.iceberg.Table icebergTable = Catalogs.loadTable(shell.getHiveConf(), properties);
+    Assert.assertEquals(PartitionSpecParser.toJson(SPEC), PartitionSpecParser.toJson(icebergTable.spec()));
+
+    // Check the HMS table parameters
+    IMetaStoreClient client = null;
+    org.apache.hadoop.hive.metastore.api.Table hmsTable;
+    try {
+      client = new HiveMetaStoreClient(metastore.hiveConf());
+      hmsTable = client.getTable("default", "customers");
+    } finally {
+      if (client != null) {
+        client.close();
+      }
+    }
+
+    Map<String, String> hmsParams = hmsTable.getParameters();
+
+    // Just check that the PartitionSpec is not set in the metadata
+    Assert.assertNull(hmsParams.get(InputFormatConfig.PARTITION_SPEC));
+
+    if (needToCheckSnapshotLocation()) {
+      Assert.assertEquals(6, hmsParams.size());
+    } else {
+      Assert.assertEquals(5, hmsParams.size());
+    }
+
+    shell.executeStatement("DROP TABLE customers");
```

Comment:
Should this be done with `DROP TABLE IF EXISTS` in a `@Before` method?

Reply:
If there is a problem with the HiveMetaHook and the `DROP TABLE` fails for whatever reason, then all of the tests would be failing. I would keep it this way until we are sure about the stability of the `DROP TABLE`. What do you think?

Reply:
I think it can fail here or in `@After` (which is what I meant to say) with the same effect. Putting it in an after method ensures that no matter what happens with the test, say a `NullPointerException` somewhere, the table is dropped for the next case.

Reply:
Does not hurt, and when we change to use a single metastore it will be good

Reply:
Yeah, this was mainly assuming that the metastore was shared.